src/test/kotlin/com/amazon/opendistroforelasticsearch/notifications/TestUtils.kt

```diff
@@ -0,0 +1,141 @@
+ * permissions and limitations under the License.
+ *
+ */
+
+package com.amazon.opendistroforelasticsearch.notifications
+
+import com.google.gson.JsonObject
+import com.google.gson.JsonParser
+import org.elasticsearch.client.Response
+import java.io.BufferedReader
+import java.io.IOException
+import java.io.InputStreamReader
+import java.nio.charset.StandardCharsets
+import java.util.Locale
+import kotlin.test.assertEquals
+
+class TestUtils {
+    companion object {
+        @Throws(IOException::class)
+        fun getResponseBody(response: Response, retainNewLines: Boolean): String {
+            val sb = StringBuilder()
+            response.entity.content.use { `is` ->
+                BufferedReader(
+                    InputStreamReader(`is`, StandardCharsets.UTF_8)).use { br ->
+                    var line: String?
+                    while (br.readLine().also { line = it } != null) {
+                        sb.append(line)
+                        if (retainNewLines) {
+                            sb.append(String.format(Locale.ROOT, "%n"))
+                        }
+                    }
+                }
+            }
+            return sb.toString()
+        }
+
+        fun jsonify(text: String): JsonObject {
+            return JsonParser.parseString(text).asJsonObject
+        }
+
+        fun verifyResponse(response: JsonObject, refTag: String, recipients: List<String>) {
+            // verify ref tag is consistent
+            val actualRefTag = response.get("refTag").asString
+            assertEquals(refTag, actualRefTag)
+
+            // verify status to each recipient
+            val actualRecipients = response.getAsJsonArray("recipients")
+            actualRecipients.forEach { item ->
+                val recipient = item.asJsonObject.get("recipient").asString
+                assert(recipients.contains(recipient))
+
+                val statusCode = item.asJsonObject.get("statusCode").asInt
+                assertEquals(200, statusCode)
+
+                val statusText = item.asJsonObject.get("statusText").asString
+                assertEquals("Success", statusText)
+            }
+        }
+    }
+
+    /** Util class to build Json entity of request body */
+    class NotificationsJsonEntity(
+        private val refTag: String?,
+        private val recipients: List<String>,
+        private val title: String?,
+        private val textDescription: String?,
+        private val htmlDescription: String?,
+        private val attachment: String?
+    ) {
+
+        var jsonEntityString: String = ""
+
+        private constructor(builder: Builder): this(builder.refTag, builder.recipients, builder.title,
+            builder.textDescription, builder.htmlDescription, builder.attachment)
+
+        fun getJsonEntityAsString(): String {
+            updateJsonEntity()
+            return jsonEntityString
+        }
+
+        private fun updateJsonEntity() {
+            jsonEntityString = String.format(
```

Comment:
probably better to use $refTag replacement instead of String.format?

Reply:
Sounds good, done!